src/posix/platform/udp.cpp

```diff
@@ -197,8 +197,6 @@ static otError receivePacket(int aFd, uint8_t *aPayload, uint16_t &aLength, otMe
 
                 memcpy(&pktinfo, CMSG_DATA(cmsg), sizeof(pktinfo));
 
-                aMessageInfo.mInterfaceId =
```

Comment:
I think we should set `aMessageInfo.mIsHostInterface` here, because when sending response to request from other host interfaces, this should be 0, that is, let the host choose the best interface.